src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -0,0 +1,91 @@
+import org.joda.time.format.ISODateTimeFormat;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class RequestByInstanceIdResource extends Resource {
+  public RequestByInstanceIdResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration(
+      "/circulation/requests/instances", router);
+
+    routeRegistration.create(this::requestByInstanceId);
+  }
+
+  private void requestByInstanceId(RoutingContext routingContext) {
+    final Result<RequestByInstanceIdRequest> requestResult =
+        RequestByInstanceIdRequest.from(routingContext.getBodyAsJson());
+
+    CompletableFuture.completedFuture(requestResult)
+      .thenApply(r -> r.map(request -> {
+        // Need to fill in the real implementation via CIRC-245.
+        // For now, we'll return some sample data.
+        final JsonObject responseBody = new JsonObject();
+        responseBody.put("id", UUID.randomUUID().toString());
+        responseBody.put("requestType", RequestType.HOLD.getValue());
+        responseBody.put("requestDate",
+            request.getRequestDate().toString(ISODateTimeFormat.dateTime()));
+        responseBody.put("requesterId", request.getRequesterId().toString());
+        responseBody.put("itemId", UUID.randomUUID().toString());
+        responseBody.put("status", "Open - Not yet filled");
+        responseBody.put("position", 1);
+
+        final JsonObject item = new JsonObject();
+        item.put("title", "Title Level Request");
+        item.put("barcode", RandomStringUtils.random(12, false, true));
+        item.put("holdingsRecordId", UUID.randomUUID().toString());
+        item.put("instanceId", request.getInstanceId().toString());
+        item.put("location", new JsonObject().put("name", "SECOND FLOOR"));
+        item.put("contributorNames", new JsonArray().add(new JsonObject().put("name", "Guy, Some")));
+        item.put("status", "Checked out");
+        item.put("callNumber", "12345678");
+
+        responseBody.put("item", item);
+
+        final JsonObject requester = new JsonObject();
+        requester.put("lastName", "Public");
+        requester.put("firstName", "John");
+        requester.put("middleName", "Q");
+        requester.put("barcode", RandomStringUtils.random(15, false, true));
+        requester.put("patronGroup", new JsonObject()
+            .put("id", UUID.randomUUID().toString())
+            .put("group", "student")
+            .put("desc", "Student"));
+        requester.put("patronGroupId", UUID.randomUUID().toString());
+
+        responseBody.put("requester", requester);
+
+        responseBody.put("fulfilmentPreference", request.getFulfilmentPreference().getValue());
+        if (request.getHoldShelfExpirationDate() != null) {
```

Comment:
I think these if statements can be replaced by using the JSON property writer helper methods

Reply:
Agreed. I made this change. Admittedly, I wasn't putting a lot of effort here due to the fact that this was mostly, if not all, throw away code.